ccm_web/client/src/components/FileUpload.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { ChangeEvent, useRef } from 'react'
+import { Button, Card, CardContent, makeStyles, Typography } from '@material-ui/core'
+import CloudUploadIcon from '@material-ui/icons/CloudUpload'
+import { useSnackbar } from 'notistack'
+
+interface FileUploadProps {
+  onUploadComplete: (file: File) => void
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    textAlign: 'center'
+  },
+  cardContent: {
+    cursor: 'pointer'
+  },
+  actionsBar: {
+    backgroundColor: '#EEEEEE'
+  },
+  actionsBarIcon: {
+    verticalAlign: 'middle'
+  },
+  input: {
+
+  },
+  uploadIcon: {
+    color: '#3F648E'
+  }
+}))
+
+function FileUpload (props: FileUploadProps): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar()
+  const handleFileSelected = (file: File|undefined): void => {
+    if (file === undefined) return
+    if (file.name.toUpperCase().match(/.+.CSV/) == null) {
+      enqueueSnackbar('File type not supported.', {
+        variant: 'error'
+      })
+      return
+    }
+    props.onUploadComplete(file)
+  }
+
+  const dragover = (e: React.DragEvent): void => {
+    e.preventDefault()
+  }
+  const dragdrop = (e: React.DragEvent): void => {
+    e.preventDefault()
+    handleFileSelected(e.dataTransfer.files[0])
+  }
+  const onFileInputClick = (): void => {
+    fileInput.current?.click()
```

Comment:
Clicking to initiate the upload doesn't work without this line, but with it this gets called 2x.
Seems like something isn't correct but it works as far as I can tell...